test/orm/test_session.py

```diff
@@ -1264,6 +1264,29 @@ def test_explicit_expunge_deleted(self):
         assert u1 not in sess
         assert object_session(u1) is None
 
+    def test_session_close_final(self):
+        users, User = self.tables.users, self.classes.User
+        self.mapper_registry.map_imperatively(User, users)
+
+        s1 = fixture_session(close_final=True)
+        u1 = User()
+
+        close_all_sessions()
+
+        assert_raises(sa.exc.InvalidRequestError, s1.add, u1)
+
+    def test_session_close_not_final(self):
+        users, User = self.tables.users, self.classes.User
+        self.mapper_registry.map_imperatively(User, users)
+
+        s1 = fixture_session(close_final=False)
```

Comment:
**Federico Caselli** (CaselIT) wrote:

Ack

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4824